lm_eval/api/task.py

```diff
@@ -568,56 +566,63 @@ def __init__(
                     key: metric_config[key]
                     for key in metric_config
                     if key
-                    not in ["metric", "aggregation", "higher_is_better", "hf_evaluate"]
+                    not in [
+                        "metric",
+                        "aggregation",
+                        "higher_is_better",
+                        "use_hf_evaluate",
+                    ]
                 }
-                hf_evaluate_metric = (
-                    "hf_evaluate" in metric_config
-                    and metric_config["hf_evaluate"] is True
+                use_hf_evaluate = (
+                    "use_hf_evaluate" in metric_config
+                    and metric_config["use_hf_evaluate"] is True
                 )
 
                 if self.config.process_results is not None:
-                    self._metric_fn_list[metric_name] = None
-                    self._metric_fn_kwargs[metric_name] = {}
+                    metric_fn = None
+                    kwargs = {}
                 elif callable(metric_name):
                     metric_fn = metric_name.__call__
                     metric_name = metric_name.__name__
-                    self._metric_fn_list[metric_name] = metric_fn
-                    self._metric_fn_kwargs[metric_name] = kwargs
                 else:
-                    self._metric_fn_list[metric_name] = get_metric(
-                        metric_name, hf_evaluate_metric
-                    )
-                    self._metric_fn_kwargs[metric_name] = kwargs
-
-                if "aggregation" in metric_config:
-                    agg_name = metric_config["aggregation"]
-                    if isinstance(agg_name, str):
-                        self._aggregation_list[metric_name] = get_aggregation(agg_name)
-                    elif callable(agg_name):  # noqa: E721
-                        self._aggregation_list[metric_name] = metric_config[
-                            "aggregation"
-                        ]
+                    assert isinstance(metric_name, str)
```

Comment:
I think we should try to minimize the amount of metric handling code that is in the task.py file if possible

Reply:
I don't have strong preference for it to be in `task.py` but currently I think atm it's the most relevant to be here since metric is attached to the task through it's config. Maybe an intermediate process between loading config and task could work?